pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -802,35 +804,49 @@ func (c *Controller) constructWorkload(pod *v1.Pod, waypoints []string, policies
 	if pod == nil {
 		return nil
 	}
-	if !IsPodReady(pod) {
+
+	// In some scenarios (e.g. outbound traffic that gates pod health checks) we must allow
```

Comment:
Seems like we have two options:

option 1 (implemented here): `{vips:[]}`, transitions to `{vips:[a,b,c]}`

option 2: `{vips:[a,b,c],state: UNREADY}` transitions to `{vips:[a,b,c],state:READY}`.

(or I guess a third is both `state` but no VIPS until healthy)

IMO the latter is preferred. It does require a small tweak on ztunnel side but that is ok. 

It is more explicit about the state of the pod - in option 1 we cannot actually tell if its healthy or not. We also are going to need more flexibility here to properly implement service (https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/1669-proxy-terminating-endpoints/README.md, etc handles terminating endpoints)

Reply:
That makes sense to me - my preference would be to 

1. merge this (as it fixes a bug which will be and already is a pretty major blocker for Normal User Workloads, for all cases we currently have).

2. Actually implement what you suggest as a followup feature - since it is not strictly required for this bugfix but we will, as you point out, need it in the future to handle other cases.

Reply:
I agree with @bleggett that it should be safe to merge what @bleggett has and he can follow up PRs on the preferred approach. This will unblock others who have pods that keeps crashing atm when adding to ambient.